### ISSUE TITLE
[BUG] Issue with `application/octet-stream` MIME type on responseBody

### ISSUE THREAD

**Issue by telday** [schemathesis/schemathesis#1134]
**Describe the bug**
When a response body is declared which specifies an `application/octet-stream` MIME type and the `type` field is set to `string` Schemathesis will throw encoding errors. All tests seemingly pass, however the run exits with code 1 indicating failure to CI. 

It looks like this is an issue with encoding binary data into UTF-8 format before sending the data to the server. 

**To Reproduce**
Create an endpoint with a request body set to the below specified YAML and run schemathesis tests against it. Note that this can be resolved by adding a `format: binary` field under the `schema` object, however this forces some occasionally unwanted behavior from generated clients. 

If possible, please post a minimal version of your API schema that cause this behavior:

```yaml
requestBody:
  required: false
  content:
    application/octet-stream:
      schema:
        type: string
```

**Expected behavior**
The test should pass as it currently does without the encoding errors and with a passing error code.

**Environment (please complete the following information):**
 - OS: Ubuntu Docker container
 - Python version: 3.8.2
 - Schemathesis version: 3.6.2
 - Spec version: 3.0

**Additional Info**
Here is an example of one of the errors outputted by Schemathesis.

```
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/schemathesis/runner/impl/core.py", line 422, in network_test
    response = _network_test(
  File "/usr/local/lib/python3.8/dist-packages/schemathesis/runner/impl/core.py", line 465, in _network_test
    response = case.call(session=session, headers=headers, timeout=timeout, verify=request_tls_verify)
  File "/usr/local/lib/python3.8/dist-packages/schemathesis/models.py", line 293, in call
    response = session.request(**data)  # type: ignore
  File "/usr/local/lib/python3.8/dist-packages/requests/sessions.py", line 542, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/local/lib/python3.8/dist-packages/requests/sessions.py", line 655, in send
    r = adapter.send(request, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/requests/adapters.py", line 439, in send
    resp = conn.urlopen(
  File "/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py", line 699, in urlopen
    httplib_response = self._make_request(
  File "/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py", line 394, in _make_request
    conn.request(method, url, **httplib_request_kw)
  File "/usr/local/lib/python3.8/dist-packages/urllib3/connection.py", line 234, in request
    super(HTTPConnection, self).request(method, url, body=body, headers=headers)
  File "/usr/lib/python3.8/http/client.py", line 1255, in request
    self._send_request(method, url, body, headers, encode_chunked)
  File "/usr/lib/python3.8/http/client.py", line 1300, in _send_request
    body = _encode(body, 'body')
  File "/usr/lib/python3.8/http/client.py", line 164, in _encode
    raise UnicodeEncodeError(
UnicodeEncodeError: 'latin-1' codec can't encode character '\u0100' in position 0: Body ('Ā') is not valid Latin-1. Use body.encode('utf-8') if you want to send it encoded in UTF-8.
```

**Comment by Stranger6667** [comment]
Thanks, @telday for reporting it :) I plan to release a fix today

**Comment by Stranger6667** [comment]
The fix is released in 3.6.3
